modules/afsocket/afsocket-source.c

```diff
@@ -1216,8 +1224,8 @@ afsocket_sd_register_stats(AFSocketSourceDriver *self)
   StatsClusterLabel labels[] =
   {
     stats_cluster_label("id", self->super.super.id),
-    stats_cluster_label("direction", "input"),
     stats_cluster_label("driver_instance", afsocket_sd_format_name(&self->super.super.super)),
+    stats_cluster_label("direction", "input"),
```

Comment:
[question]

Could you kindly explain why you have changed the order of the labels?

To the stats API the same labels in different orders are different labels. This means that the call site has to make sure that it sets up the labels in the right order. Although you have changed the order on both the registration and unregistration site, it is not clear to me why we would ever deviate from setting them up in alphabetical order for hard coded labels. If we do not have a globally accepted convention for the order of hard-coded labels, we could easily make mistakes, IMO. Prometheus will sort the labels in the end either way, if I remember correctly.

Reply:
It looked better (and we haven't released this metric yet).
We already have non-alphabetically-sorted counters, as I always preferred listing `id` first.

I think mistakes could be avoided if we always used the key builder for shared counters, which does sort labels.
For local-only uses, it is relatively easy to keep the same order by extracting a function that creates the key for registration and deregistration.